src/workshed.ts

```diff
@@ -0,0 +1,390 @@
+  visitUrl("main.php");
+
+  const expected = pieces.join(",");
+
+  if (expected !== get("trainsetConfiguration", "")) {
+    throw new Error(
+      `Expected trainset configuration to have changed, expected "${expected}" but instead got ${get(
+        "trainsetConfiguration",
+        ""
+      )}`
+    );
+  }
+}
+
+class GarboWorkshed {
+  workshed: Item;
+  done: () => boolean;
+  action: () => void;
+
+  constructor(workshed: Item, done: () => boolean, action: () => void) {
+    this.workshed = workshed;
+    this.done = done;
+    this.action = action;
+  }
+}
+
+export function currentWorkshed(): GarboWorkshed {
+  return worksheds.find((workshed) => workshed.workshed === getWorkshed()) ?? defaultWorkshed;
+}
+
+export function getGarboWorkshed(item: Item): GarboWorkshed {
+  return worksheds.find((workshed) => workshed.workshed === item) ?? defaultWorkshed;
+}
+
+const trainset = new GarboWorkshed(
+  // eslint-disable-next-line libram/verify-constants
+  $item`model train set`,
+  () => {
+    // We should always get value from the trainset, so we would never switch from it
+    return false;
+  },
+  () => {
+    const pieces = [
+      TrainsetPiece.DOUBLE_NEXT_STATION,
+      TrainsetPiece.GAIN_MEAT,
+      TrainsetPiece.CANDY,
+      TrainsetPiece.RANDOM_BOOZE,
+      TrainsetPiece.DROP_LAST_FOOD_OR_RANDOM,
+      TrainsetPiece.ORE,
+      TrainsetPiece.EFFECT_MP,
+      TrainsetPiece.GAIN_STATS,
+    ];
+    if (!isTrainsetConfigurable()) return;
+    else if (pieces.join(",") === get("trainsetConfiguration", "")) return;
+    setTrainsetConfiguration(pieces);
+  }
+);
+
+const CMC = new GarboWorkshed(
+  $item`cold medicine cabinet`,
+  () => {
+    return get("_coldMedicineConsults") >= 5;
+  },
+  () => {
+    if (property.getNumber("_nextColdMedicineConsult") > totalTurnsPlayed()) return;
+    const options = visitUrl("campground.php?action=workshed");
+    let i = 0;
+    let match;
+    const regexp = /descitem\((\d+)\)/g;
+    const itemChoices = new Map<Item, number>();
+    if (!globalOptions.noBarf) {
+      // if spending turns at barf, we probably will be able to get an extro so always consider it
+      itemChoices.set($item`Extrovermectin™`, -1);
+    }
+
+    while ((match = regexp.exec(options)) !== null) {
+      i++;
+      const item = descToItem(match[1]);
+      itemChoices.set(item, i);
+    }
+
+    const bestItem = maxBy([...itemChoices.keys()], garboValue);
+    const bestChoice = itemChoices.get(bestItem);
+    if (bestChoice && bestChoice > 0) {
+      visitUrl("campground.php?action=workshed");
+      runChoice(bestChoice);
+    }
+    if (handlingChoice()) visitUrl("main.php");
+  }
+);
+
+const pizzaCube = new GarboWorkshed(
+  $item`diabolic pizza cube`,
+  () => {
+    return dietCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const asdonMartin = new GarboWorkshed(
+  $item`Asdon Martin keyfob`,
+  () => {
+    return haveEffect($effect`Driving Observantly`) >= estimatedTurns();
```

Comment:
We will want to deliberately overbuff for at least 1 extra day unless you're ascending

Reply:
To be specific, he means changing the `>=` to a `>` unless ascending is true

Reply:
That's not actually quite what I mean!

I mean that if someone's in long aftercore, the best behavior for asdon isn't going to be "buff up until the end of the day, then switch to cmc", it's going to be "buff up for a week, then switch between cmc and trainrealm until you need asdon again". And even if you aren't in long aftercore, on your non-ascending leg you'll want enough asdon turns for that day and the next.

It's not obvious what best behavior is, but I'd say at least two days of turns if you aren't ascending, maybe a pref or an argument for doing more.

Reply:
What would be the `estimatedTurns()` for the extra day? Something like `400 + clamp((VOA-4000)/10, 0, 300)`?